File: source_funcs/go_transcoder/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N.go
package main

func f_gold(n int) int {
	var result int = 0
	for i := int(0); i <= n; i++ {
		for j := int(0); j <= n-i; j++ {
			for k := int(0); k <= (n - i - j); k++ {
				if i+j+k == n {
					result++
				}
			}
		}
	}
	return result
}

File: rust_funcs/go_transcoder/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(n: i32) -> i32 { 
    let mut result = 0;
    for i in 0..=n {
        for j in 0..=(n - i) {
            for k in 0..=(n - i - j) {
                if i + j + k == n {
                    result += 1;
                }
            }
        }
    }
    result
}
